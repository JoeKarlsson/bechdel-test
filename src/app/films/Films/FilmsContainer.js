import React, { Component } from 'react';
import Films from './Films';
import api from '../../helper/api';

class FilmsContainer extends Component {
	constructor() {
		super();
		this.state = {
			films: [],
			loading: false,
		};
		this.getAllFilms = this.getAllFilms.bind(this);
	}

	componentDidMount() {
		this.getAllFilms();
	}

	getAllFilms() {
		const url = '/api/film';
		const options = {
			method: 'GET',
		};

		this.setState({
			loading: true,
		});

		api(url, options)
			.then(data => {
				this.setState({
					films: data,
					loading: false,
				});
			})
			.catch(err => {
				console.err(err);
				this.setState({
					loading: false,
				});
			});
	}

	render() {
		return <Films {...this.state} />;
	}
}

FilmsContainer.propTypes = {};

FilmsContainer.defaultProps = {};

export default FilmsContainer;
