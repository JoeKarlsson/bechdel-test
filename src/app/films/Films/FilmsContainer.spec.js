import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilmsContainer from './FilmsContainer';

jest.mock('../../helper/api');

configure({ adapter: new Adapter() });

describe('FilmsContainer', () => {
	let wrapper;
	let inst;

	beforeEach(() => {
		wrapper = shallow(
			<MemoryRouter>
				<FilmsContainer />
			</MemoryRouter>
		);
		inst = wrapper.instance();
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('match the snapshot', () => {
				const component = renderer.create(
					<MemoryRouter>
						<FilmsContainer />
					</MemoryRouter>
				);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});
			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});
			it('should not have any inital props', () => {
				const initialProps = inst.props;
				const expectedProps = {
					children: <FilmsContainer />,
				};
				expect(initialProps).toMatchObject(expectedProps);
			});
			it('should not have any inital state', () => {
				const initialState = inst.state;
				const expectedState = {};
				expect(initialState).toMatchObject(expectedState);
			});
		});
	});
});
