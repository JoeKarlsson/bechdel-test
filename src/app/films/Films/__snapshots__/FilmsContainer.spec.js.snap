// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsContainer rendering initial state match the snapshot 1`] = `
<div>
  No Films Have Been Added Yet
</div>
`;
